lib/encoder/include/Pololu37D.h

```diff
@@ -21,7 +21,16 @@ class Pololu37D final : public Encoder {
   bool getAngularVelocityDegPerSec(float &speed) override;
   bool reset() override;
 
+  bool read();
```

Comment:
This function call is a bit vague. We have a call to get angle degree and one for angular velocity. However I don't feel that "read" tells the user anything about the call.

Reply:
Yea i agree that it is a bit vague. I was trying to mirror the functionality used in the AEAT6012 driver.

Reply:
I will try to improve readabilityty

Reply:
lol *readability that was ironic

Reply:
Discussed offline, but we'll have an encoder interface refactor after SAR to separate the read from the getAngle and getAngularVelocity funcitons, so the latter two only returns the stored member vars while read should be called periodically in a separate thread to update those member vars. Looks like this driver already implements it, but the current controller code doesn't use this approach

Reply:
@Yehia-Fahmy I think you should revert to the old method for this driver for now, so keep read() private and call it within the getAngle() and getAngularVelocity() functions. Then after SAR, change all encoder libraries to use the new method

Reply:
currently getAngle and getVelocity call the read function